app/common/directives/modal-container.directive.js

```diff
@@ -35,12 +35,26 @@ function ModalContainer($timeout, $rootScope, $compile, ModalService) {
         ModalService.onClose(closeModal, $scope);
 
         //var classChangePromise = null;
+        function makeTemplateScope(scope) {
+            // If no scope was passed, create a child of our directive scope
+            if (!scope) {
+                // @todo should this just be $rootScope.$new() ??
+                return $scope.$new();
+            }
+
+            // If scope isn't actually a scope yet, make it into one
+            if (scope.constructor !== $rootScope.constructor) {
+                return angular.extend($rootScope.$new(), scope);
```

Comment:
@willdoran I'm particularly pleased how this worked out. Take all the load of managing scope off callers to ModalService.. they can now just pass plain JS objects.
Might be worth doing the same for sliders too..

Only :( is the lack of tests